src/lib/contracts/contractFetcher.ts

```diff
@@ -81,29 +109,93 @@ export const contractFetcher =
           handleFallback(resolve, reject, e);
         });
 
+      const isThroughMulticall = ethers.isAddress(arg0);
+      const isUrgent = priority === "urgent";
+
+      let timeout = 2000;
+
+      if (isThroughMulticall) {
+        if (isUrgent) {
+          const workerStartupTimeout = 1000;
+          timeout = timeout + workerStartupTimeout;
+        } else {
+          const workerTimeout = 5000;
+          const mainThreadTimeout = 5000;
```

Comment:
logic makes sense to me
just thinking 
- what if we do all urgent calls in main thread. that way we would save this startup timeout time. Not a priority for this iteration but rather a point to think
- code is self explaining but should we put numbers into constants? maybe not, not sure really.

Reply:
I think this place should be changed in part 2 as we plan to change the fallback logic

Reply:
If we do urgent all urgen calls in main thread we will encouter more stuttering during page load. as the first render causes a lot of simultanious urgent calls

Reply:
moved to constant. but the archietcture will change i hope